Control/DeepSeq.hs

```diff
@@ -202,6 +215,87 @@ f $!! x = x `deepseq` f x
 force :: (NFData a) => a -> a
 force x = x `deepseq` x
 
+newtype UnitLTR = UnitLTR {getUnitLTR :: ()}
+
+#if MIN_VERSION_base(4,9,0)
+instance Semi.Semigroup UnitLTR where
+  UnitLTR () <> y = y
+#endif
+
+instance Monoid UnitLTR where
+  mempty = UnitLTR ()
+  UnitLTR () `mappend` y = y
```

Comment:
To avoid invoking the wrath of `-Wnoncanonical-monoid-instances` in later GHCs, it'd be prudent to define `mappend` like this:

```haskell
instance Monoid UnitLTR where
  mempty = UnitLTR ()
#if MIN_VERSION_base(4,9,0)
  mappend = (<>)
#else
  UnitLTR () `mappend` y = y
#endif
```

And similarly for `UnitRTL`.